src/cryptonote_core/service_node_list.cpp

```diff
@@ -2977,6 +2978,29 @@ namespace service_nodes
     return crypto::null_pkey;
   }
 
+  template<typename Iter, typename RandomGenerator>
+  Iter select_randomly(Iter start, Iter end, RandomGenerator& g) {
+      std::uniform_int_distribution<> dis(0, std::distance(start, end) - 1);
+      std::advance(start, dis(g));
+      return start;
+  }
+
+  template<typename Iter>
+  Iter select_randomly(Iter start, Iter end) {
+      static std::random_device rd;
+      static std::mt19937 gen(rd());
```

Comment:
We have a pre-seeded global rng available (as `tools::rng`), accessible via "common/random.h".